<?xml version="1.0"?>
<launch>

    <arg name="camera_type" value="realsense"/>
    <!-- The input reference frames -->
    <arg name="eye_on_hand"           default="true" />
    <arg name="robot_base_frame"      default="base_link" />
    <arg name="robot_effector_frame"  default="ee_link" />
    <arg name="tracking_base_frame"   default="zivid_optical_frame" />
    <arg name="tracking_marker_frame" default="zivid_camera_aruco_0" />

    <arg name="ns" default="zivid_camera"/>
    <arg name="rotation_flag" default="100"/>

    <node name="zivid_camera" pkg="zivid_camera" type="zivid_camera_node" ns="zivid_camera">
        <param name="use_save"       value="false"/>
        <param name="save_directory" value=""/>
    </node>
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>
    <node name="zivid_sample" pkg="zivid_camera" type="zivid_continuous_capture.py"/>    

    <arg name="image"         default="color/image_color"/>
    <arg name="markers"       default="aruco"/>
    <arg name="depth_image"   default="depth/image_raw"/>
    <arg name="camera_info"   default="depth/camera_info"/>
    <arg name="display"       default="true"/>
    <arg name="is_depth_in_meters" default="true" />

    <group ns="$(arg ns)">
        <node name="depth_detector_node" pkg="aruco_detector" type="depth_detector_node" output="screen">
          <param name="image"         value="$(arg image)" />
          <param name="markers"       value="$(arg markers)"/>
          <param name="depth_image"   value="$(arg depth_image)" />
          <param name="camera_info"   value="$(arg camera_info)" />
          <param name="tf_prefix"     value="$(arg ns)" />
          <param name="display"       value="$(arg display)"/>
          <param name="is_depth_in_meters" type="string" value="$(arg is_depth_in_meters)" />
        </node>
     </group>

    <!-- Run the ViSP solver to determine the extrinsic parameters -->
    <node name="hand_eye_solver"
          pkg="visp_hand2eye_calibration"
          type="visp_hand2eye_calibration_calibrator">
        <remap from="/compute_effector_camera_quick" to="compute_effector_camera_quick"/>
    </node>

    <!-- <node name="dummy_handeye" pkg="tf" type="static_transform_publisher" args="0 0 0.0 0 0 0 $(arg robot_effector_frame) $(arg tracking_base_frame) 10" /> -->
    
    <node name="handeye_calibration_node" pkg="cares_hand_eye_calibration" type="handeye_calibration_node.py" output="screen">
        <param name="eye_on_hand"             value="$(arg eye_on_hand)"/>
        <param name="robot_base_frame"        value="$(arg robot_base_frame)" />
        <param name="robot_effector_frame"    value="$(arg robot_effector_frame)" />
        <param name="tracking_base_frame"     value="$(arg tracking_base_frame)" />
        <param name="tracking_marker_frame"   value="$(arg tracking_marker_frame)" />
        <param name="rotation_flag"           value="$(arg rotation_flag)" />
    </node>

    <!-- Setup rviz -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find cares_hand_eye_calibration)/rviz/arm-mapping.rviz" />

</launch>